### PR TITLE
implemented light/dark mode, used react context

### PR DESCRIPTION
**Pull request by GedasIncirauskas**
1. Created light/dark mode switcher.
2. Implemented light/dark mode functionality to app.
3. Used react context to pass theme value.
4. Added functions to save theme value to localStorage.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+import { Switch, Route, HashRouter } from 'react-router-dom';
 import { Header, Footer } from './components';
 import { Home, About, Contact, CityWeather } from './pages';
+import { DarkModeContext } from './contexts/darkMode';
+import { GlobalStyles } from './styles/globalStyles';
 import * as S from './App.styles';
 
 const App = () => {
+  const darkModeContext = useContext(DarkModeContext);
+  const { isDarkMode } = darkModeContext;
```
Comment (JuliaShavyrova) [inline src/App.jsx]: lines can be merged into one

Comment (GedasIncirauskas) [reply src/App.jsx]: Thanks, fixed.

Comment (JuliaShavyrova) [reply src/App.jsx]: check other places where code also can be simplified

Comment (GedasIncirauskas) [reply src/App.jsx]: Already fixed at both places where I am using it :)

**Review by JuliaShavyrova — APPROVED** [review]

**Pull request merged**